include/ensmallen_bits/indicators/epsilon.hpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file epsilon.hpp
+ * @author Rahul Ganesh Prabhu
+ * @author Nanubala Gnana Sai
+ *
+ * Epsilon indicator
+ * A binary quality indicator that is capable of detecting whether one
+ * approximation set is better than another.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_INDICATORS_EPSILON_HPP
+#define ENSMALLEN_INDICATORS_EPSILON_HPP
+
+namespace ens {
+
+  /**
+   * The epsilon indicator is a binary quality indicator that was proposed by
+   * Zitzler et. al. in response to the observation that quality indicators could
+   * only give information that one approximation set is not worse than another.
+   * The binary epsilon indicator, however, is capable of detecting if an
+   * approximation is better than another.
+   *
+   * \f[ I_c(A,B) = \max_{z_2 \in B} \
+   *                \min_{z_1 \in A} \
+   *                \max_{1 \leq i \leq n} \ \frac{z_1}{z_2}\
+   *                \f]
+   *
+   * For more information, please see:
+   *
+   * @code
+   * @article{1197687,
+   *    author   = {E. Zitzler and L. Thiele and M. Laumanns and C. M. Fonseca and
+   *                V. G. da Fonseca},
+   *    title    = {Performance assessment of multiobjective optimizers: an
+   *                analysis and review},
+   *    journal  = {IEEE Transactions on Evolutionary Computation},
+   *    year     = {2003},
+   * }
+   * @endcode
+   */
+  class Epsilon
+  {
+    /**
+     * Find the epsilon value of the front with respect to the given reference
+     * front.
+     *
+     * @param front The given approximation front.
+     * @param referenceFront The given reference front.
+     * @return The epsilon value of the front.
+     */
+    double Indicate(arma::cube& front,
```

Comment:
Is Indicate really the word we're going for? I'd suggest EvaluateFront / Evaluate or something like that. @zoq Your thoughts?

Reply:
`Evaluate` sounds good, also do you mind to put this into an existing optimizer, this is what we did for policies as well, and I'd like to keep the root folder clean.

Reply:
I suppose you mean I should put this into optimizers similar to how ```sgd``` has ```decay_policies```? If I'm correct till here, I believe that's the wrong way to go. An indicator is essentially a metric, so I was following the conventions as is provided [here](https://www.mlpack.org/doc/stable/doxygen/lmetric_8hpp_source.html), notice ```lmetric.hpp``` is inside ```metric/```, I'm going for something similar :D.

Reply:
> `Evaluate` sounds good, also do you mind to put this into an existing optimizer, this is what we did for policies as well, and I'd like to keep the root folder clean.

This is actually genius! So, maybe we could do 
```c++
<MatType, ...ArbitraryFunctionType, IndicatorType = ens::HyperVolume<MatType>> 
typename MatType::elem_type Optimize()
{
/*code*/
return IndicatorType.Evaluate(bestFront);
}
```
This way Optimize would return a single value as per our choice of metric! Perfect!
Then we could have ```indicator_policy/``` for nsga2 folder and others. But I would need to implement the same indicator code for each and every MOO optimizer, so that's the bad part. Am I following what you said correctly?

Reply:
Oh wait, I would need  reference front .... this won't work :(

Reply:
Okay, as discussed, I will move this to utility/indicators and make it something similar to lmetric.hpp.